lib/ide/parser/src/api.rs

```diff
@@ -1,20 +1,57 @@
 use crate::prelude::*;
+use ast::ID;
+use std::ops::Sub;
 
-pub type Ast = ast::Ast;
+use serde::Serialize;
+use serde::Deserialize;
 
+pub type Ast = ast::Ast;
 
 
 // ============
 // == Parser ==
 // ============
 
+#[derive(Clone, Debug, Default, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
```

Comment:
Please remove spaces between arguments. (same for other instances)

Reply:
BTW, if you add new types, there should be new sections for them. `Index` is not a "Parser".